### PR TITLE
Fix Cohere CI

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Currently some tests are failing, see 

https://github.com/huggingface/transformers/actions/runs/9360271371/job/25765482477

This PR skips 2 of them and make 4 pass: 

```
FAILED tests/models/cohere/test_modeling_cohere.py::CohereModelTest::test_eager_matches_sdpa_generate - ValueError: Some modules are dispatched on the CPU or the disk. ..
FAILED tests/models/cohere/test_modeling_cohere.py::CohereIntegrationTest::test_batched_4bit - torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 88.00 MiB. GPU
FAILED tests/models/cohere/test_modeling_cohere.py::CohereIntegrationTest::test_batched_small_model_logits - RuntimeError: Half did not match Float
FAILED tests/models/cohere/test_tokenization_cohere.py::CohereTokenizationTest::test_torch_encode_plus_sent_to_model - RuntimeError: scaled_dot_product_attention_flash_attention does not support FP16 on the platforms without amx_fp16 support
```

### CODE REVIEW

File: tests/models/cohere/test_modeling_cohere.py [hunk 1]
```diff
@@ -291,6 +291,14 @@ def setUp(self):
         self.model_tester = CohereModelTester(self)
         self.config_tester = ConfigTester(self, config_class=CohereConfig, hidden_size=37)
 
+    @unittest.skip("foo")
+    def test_initialization(self):
+        super().test_initialization()
+
+    @unittest.skip("foo")
+    def test_fast_init_context_manager(self):
+        super().test_fast_init_context_manager()
```
Comment (ydshieh) [inline tests/models/cohere/test_modeling_cohere.py:300]: I will open issue to keep this track, but let's follow what we have discussed earlier that we sometimes also need to skip some tests.

File: tests/models/cohere/test_modeling_cohere.py [hunk 2]
```diff
@@ -373,10 +381,10 @@ def test_batched_4bit(self):
 
         EXPECTED_TEXT = [
             'Hello today I am going to show you how to make a simple and easy card using the new stamp set called "Hello" from the Occasions catalog. This set is so versatile and can be used for many occasions. I used the new In',
-            "Hi there, here we are again with another great collection of free fonts. This time we have gathered 10 free fonts that you can download and use in your designs. These fonts are free for personal and commercial use. So",
+            "Hi there, here we are again with another great collection of free fonts for your next project. This time we have gathered 10 free fonts that you can download and use in your designs. These fonts are perfect for any kind",
         ]
 
-        model = CohereForCausalLM.from_pretrained(model_id)
+        model = CohereForCausalLM.from_pretrained(model_id, device_map="auto")
```
Comment (ydshieh) [inline tests/models/cohere/test_modeling_cohere.py:387]: This was previously GPU OOM. After changing to `auto`, I have to update the expected output value too.

File: tests/models/cohere/test_tokenization_cohere.py [hunk 3]
```diff
@@ -46,6 +46,11 @@ def get_rust_tokenizer(self, **kwargs):
         kwargs.update(self.special_tokens_map)
         return CohereTokenizerFast.from_pretrained(self.tmpdirname, **kwargs)
 
+    # This gives CPU OOM on a single-gpu runner (~60G RAM). On multi-gpu runner, it has ~180G RAM which is enough.
+    @require_torch_multi_gpu
+    def test_torch_encode_plus_sent_to_model(self):
+        super().test_torch_encode_plus_sent_to_model()
```
Comment (ydshieh) [inline tests/models/cohere/test_tokenization_cohere.py:52]: CPU OOM

File: tests/models/cohere/test_modeling_cohere.py [hunk 4]
```diff
@@ -291,6 +291,14 @@ def setUp(self):
...
         self.config_tester = ConfigTester(self, config_class=CohereConfig, hidden_size=37)
 
+    @unittest.skip("Failing.")
+    def test_initialization(self):
+        super().test_initialization()
+
+    @unittest.skip("Failing.")
+    def test_fast_init_context_manager(self):
+        super().test_fast_init_context_manager()
+
```
Comment (ydshieh) [inline tests/models/cohere/test_modeling_cohere.py:301]: I will open issue to track this, but as we discussed earlier, let's not spend too much time on some tests and focus on other priorities.

File: .github/workflows/self-pr-slow-ci.yml [hunk 5]
```diff
@@ -110,7 +110,10 @@ jobs:
 
       - name: Run all tests on GPU
         working-directory: /transformers
-        run: python3 -m pytest -v -rsfE --make-reports=${{ matrix.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports tests/${{ matrix.folders }}
+        run: |
+          export CUDA_VISIBLE_DEVICES="$(python3 utils/set_cuda_devices_for_ci.py --test_folder ${{ matrix.folders }})"
+          echo $CUDA_VISIBLE_DEVICES
+          python3 -m pytest -v -rsfE --make-reports=${{ matrix.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports tests/${{ matrix.folders }}
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply .github/workflows/self-pr-slow-ci.yml:116]: It requires a new script below (see at the end).

**Review by LysandreJik — APPROVED** [review]
Ok, thanks Yih-Dar!

**Pull request merged**